sound/soc/sof/ipc4-topology.c

```diff
@@ -17,6 +17,11 @@
 #include "ipc4-topology.h"
 #include "ops.h"
 
+static bool force_cpc_to_zero;
```

Comment:
Commit messsage:

Add a new module parameter ipc4_force_cpc_to_zero which can be used to
force the kernel to ignore the queried CPC value for the given module
and send 0 instead.

"module" has two meanings in this sentence, we are talking about kernel module and firmware module, not the same thing.

Also this ignores ALL values for ALL firmware modules, there is no selectivity or granularity as hinted by "the given module".

The flag is intended to be used only when there is a need to validate the
firmware behavior regarding to clock scaling since the 0 CPC value will
force the DSP to run in full speed.

this patch does the exact opposite, it prevents the DSP from using clock scaling, it does not validate anything related to clock scaling.

Reply:
@plbossart, let me reword. It provides validation for clock scaling by disabling to provide additional data point to rule out clock scaling caused failure.